src/Decompiler/Analysis/SsaTransform.cs

```diff
@@ -27,164 +27,893 @@
+                    {
+                        var fpuUseExpr = arch.CreateFpuStackAccess(
+                            ssa.Procedure.Frame,
+                            fpuUse.FpuStackOffset,
+                            PrimitiveType.Word64); //$TODO: datatype?
+                        fpuUseExpr = fpuUseExpr.Accept(this);
+                        ci.Uses.Add(
+                            new CallBinding(
+                                stgUse,
+                                fpuUseExpr));
+                    }
+                    else
+                    {
+                        var arg = stgUse.Accept(ab);
+                        arg = arg.Accept(this);
+                        ci.Uses.Add(new CallBinding(stgUse, arg));
+                    }
+                }
+
+                if (calleeFlow.TerminatesProcess)
+                {
+                    // We just discovered this basic block doesn't terminate.
+                    this.blockstates[block].terminates = true;
+                    return;
+                }
+
+                foreach (var def in calleeFlow.Trashed.Where(dd => !(dd is FpuStackStorage)))
+                {
+                    var d = ssa.Procedure.Frame.EnsureIdentifier(def);
+
+                    //$HACK: special case for FPU stacks; is there a cleaner way
+                    // to do this?
+                    if (def == arch.FpuStackRegister && calleeFlow.Constants.ContainsKey(def))
+                    {
+                        // We will add an explicit increment statement later.
+                        continue;
+                    }
+                    ci.Definitions.Add(
+                        new CallBinding(
+                            def,
+                            NewDef(d, ci.Callee, false)));
+                }
+                foreach (var de in calleeFlow.grfTrashed)
+                {
+                    var grfs = arch.GetFlagGroup(de.Key, de.Value);
+                    foreach (var bit in grfs.GetFlagBitMasks())
+                    {
+                        var grf = arch.GetFlagGroup(grfs.FlagRegister, bit);
+                        var d = ssa.Procedure.Frame.EnsureFlagGroup(grf);
+                        ci.Definitions.Add(
+                            new CallBinding(
+                                grf,
+                                NewDef(d, ci.Callee, false)));
+                    }
+                }
+                //$REVIEW: this is very x86/x87 specific; find a way to generalize
+                // this to any sort of stack-based discipline.
+                foreach (var def in calleeFlow.Trashed.OfType<FpuStackStorage>()
+                    .Where(def => def.FpuStackOffset >= fpuStackDelta))
+                {
+                    var fpuDefExpr = arch.CreateFpuStackAccess(
+                        ssa.Procedure.Frame, 
+                        def.FpuStackOffset,
+                        PrimitiveType.Word64); //$TODO: datatype?
```

Comment:
@uxmal This code breaks assumption that all definitions of `CallInstruction` are `Identifiers` 
So code like that https://github.com/uxmal/reko/blob/d97a894bef758e15beba2f4e8e2af629e97d9676/src/Decompiler/Analysis/TrashedRegisterFinder3.cs#L403 will cause exception.
Possible solution is https://github.com/ptomin/reko/commit/ebcb3bab6dda35bf16bdbbbcc1e380208a383ea7

Reply:
All tests (except 3) passed. New output for that failed tests you can see in 0a1a91f.
And it gives appropriate output on `Blade.dcproject`

Reply:
Also I have changed `UsedRegisterFinder` in 0a1a91f  to make live-out computation work correctly

Reply:
I'll raise PR(s) this weekend